File: project/src/main/java/producers/youtubeProducer/YoutubeData.java
package producers.youtubeProducer;

import producers.BaseDTO;

public class YoutubeData extends BaseDTO {


    private final String videoTitle;
    private final String videoLink;
    private final String imageLink;

    public YoutubeData(int plz, String locationName, String videoTitle, String videoLink, String imageLink)
    {
        super(plz, "youtube", locationName);
        this.videoTitle = videoTitle;
        this.videoLink = videoLink;
        this.imageLink = imageLink;
    }

    public String getVideoTitle() {
        return videoTitle;
    }

    public String getVideoLink() {
        return videoLink;
    }

    public String getImageLink() {
        return imageLink;
    }
}

File: project_webclient/webclient/src/app/youtube/youtubeData.ts
/** A class containing youtube video data. */
export class YoutubeData {
    /**
     * The title of the video.
     */
    videoTitle: String;
    /**
     * The link to the preview image.
     */
    imageLink: String;
    /**
     * The link to the video.
     */
    videoLink: String;
    /**
     * The name of the location this data is for.
     */
    locationName: String;
    /**
     * The creation date of this data.
     */
    creationDate: String;
    /**
     * The "Postleitzahl" or zip code for this data.
     */
    plz: String;

    /**
     * Constructor for the youtube data class.
     * 
     * @param plz The "Postleitzahl" or zip code for this data.
     * @param creationDate The creation date of this data.
     * @param locationName The name of the location this data is for.
     * @param videoLink The link to the video.
     * @param imageLink The link to the preview image.
     * @param videoTitle The title of the video.
     */
    constructor(plz: String, creationDate: String, locationName: String, videoLink: String, imageLink: String, videoTitle: String ){
        this.plz = plz;
        this.creationDate = creationDate;
        this.locationName = locationName;
        this.videoLink = videoLink;
        this.imageLink = imageLink;
        this.videoTitle = videoTitle;
    }
}
